pgjdbc/src/test/java/org/postgresql/test/jdbc2/ParameterStatusTest.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2019, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.test.jdbc2;
+
+import org.postgresql.PGConnection;
+import org.postgresql.core.ServerVersion;
+import org.postgresql.test.TestUtil;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.sql.Connection;
+import java.sql.Statement;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.logging.Logger;
+
+//
+// This test covers client interface for server ParameterStatus messages
+// (GUC_REPORT) parameters via PGConnection.getParameterStatuses() and
+// PGConnection.getParameterStatus() .
+//
+public class ParameterStatusTest extends BaseTest4 {
+
+  private final TimeZone tzPlus0800 = TimeZone.getTimeZone("GMT+8:00");
+  private final Logger logger = Logger.getLogger(ParameterStatusTest.class.getName());
+
+  private Connection con;
+
+  @Test
+  public void expectedInitialParameters() throws Exception {
+    TimeZone.setDefault(tzPlus0800);
+    con = TestUtil.openDB();
+
+    Map<String,String> params = ((PGConnection)con).getParameterStatuses();
+
+    // PgJDBC forces the following parameters
+    Assert.assertEquals("UTF8", params.get("client_encoding"));
+    Assert.assertNotNull(params.get("DateStyle"));
+    Assert.assertTrue(params.get("DateStyle").startsWith("ISO"));
```

Comment:
Please refrain from these kind of asserts. If it fails it will be really hard to debug.
The failure would look like: `expected true, got false`, however in fact we would be interested in actual value of `DateStyle` parameter

Reply:
Good point. I'll change it to use `assertThat`